baremetal_network_provisioning/drivers/hp/bm_net_provisioning_driver.py

```diff
@@ -202,3 +185,16 @@ def _create_port(self, port):
                 LOG.error(_LE("Physical switch is not Enabled '%s' "),
                           bnp_switch.status)
                 self._raise_ml2_error(wexc.HTTPBadRequest, 'create_port')
+
+    def _protocol_driver(self, bnp_switch):
+        """Get the protocol driver instance based on protocol."""
+        protocol_type = bnp_switch.access_protocol
+        snmp_protocol = constants.PROTOCOL_SNMP
+        try:
+            if constants.PROTOCOL_SNMP in protocol_type:
+                driver = self.protocol_manager.protocol_driver(snmp_protocol)
```

Comment:
why are you separating snmp here? protocol_manager.protocol_driver(protocol_type) wouldn't work?